kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -169,7 +169,9 @@ public void commit() throws IOException {
       log.debug("Upload complete for bucket '{}' key '{}'", bucket, key);
     } catch (IOException e) {
       log.error("Multipart upload failed to complete for bucket '{}' key '{}'", bucket, key);
-      throw new DataException("Multipart upload failed to complete.", e);
+      throw new RetriableException(
+          String.format("Multipart upload failed to complete: %s", e.getMessage())
+      );
```

Comment:
lets be careful here that we don't overwrite this PR #378

not sure we want to retry this. keep this as `DataException`

this is in the `commit()`, shouldn't affect `write()`

Reply:
The reason I changed this was based on a previous discussion with @kkonstantine that we should only have a `DataException` when the data has an issue. This looked more like an issue that is not caused by the data in the record itself. The approach may have changed since your PR so let's double check.

Reply:
Instead of DataException, we could also use an exception class that is not retriable but more appropriate for the cause?

Reply:
why not a `ConnectException` then?

Reply:
That works for me 👍 The naming `testPropagateRetriableS3Exceptions` from PR #378 tests made me think we consider it retriable.

Reply:
this case will fail the connector so lets keep it that way. That test tests something else.